ietfYangDraftPull/draftPullUtility.py

```diff
@@ -0,0 +1,186 @@
+        for f2 in os.listdir(directory):
+            # Same module name ?
+            if f2.split('.yang')[0].split('@')[0] == module_name:
+                if f2.split(module_name)[1].startswith('.') or f2.split(module_name)[1].startswith('@'):
+                    files_to_delete.append(f2)
+                    revision = f2.split(module_name)[1].split('.')[0].replace('@', '')
+                    if revision == '':
+                        yang_file_path = '{}/{}'.format(directory, f2)
+                        revision = get_latest_revision(os.path.abspath(yang_file_path), LOGGER)
+                        if revision is None:
+                            continue
+                    try:
+                        # Basic date extraction can fail if there are alphanumeric characters in the revision filename part
+                        year = int(revision.split('-')[0])
+                        month = int(revision.split('-')[1])
+                        day = int(revision.split('-')[2])
+                        revisions.append(datetime(year, month, day))
+                    except Exception:
+                        LOGGER.error('Failed to process revision for {}: (rev: {})'.format(f2, revision))
+                        if month == 2 and day == 29:
+                            revisions.append(datetime(year, month, 28))
+                        else:
+                            continue
+        # Single revision...
+        if len(revisions) == 0:
+            continue
+        # Keep the latest (max) revision and delete the rest
+        latest = revisions.index(max(revisions))
+        files_to_delete.remove(files_to_delete[latest])
+        for fi in files_to_delete:
+            if 'iana-if-type' in fi:
+                break
+            os.remove('{}/{}'.format(directory, fi))
+
+
+def get_draft_module_content(ietf_draft_url: str, experimental_path: str, LOGGER):
+    """ Update download links for each module found in IETFDraft.json and try to get their content.
+
+    Aruments:
+        :param ietf_draft_url       (str) URL to private IETFDraft.json file
+        :param experimental_path    (str) full path to the cloned experimental modules
+        :param LOGGER               (obj) formated logger with the specified name
+    """
+    ietf_draft_json = {}
+    response = requests.get(ietf_draft_url)
+    if response.status_code == 200:
+        ietf_draft_json = response.json()
+    for key in ietf_draft_json:
+        with open('{}/{}'.format(experimental_path, key), 'w+') as yang_file:
+            yang_download_link = ietf_draft_json[key][2].split('href="')[1].split('">Download')[0]
+            yang_download_link = yang_download_link.replace('new.yangcatalog.org', 'yangcatalog.org')
+            try:
+                yang_raw = requests.get(yang_download_link).text
+                yang_file.write(yang_raw)
+            except:
+                LOGGER.warning('{} - {}'.format(key, yang_download_link))
```

Comment:
This one is ok since we don t care about the error in here and we just move on. exception will log it as error and we want only warning in here.